src/StoragelessSession/Http/SessionMiddleware.php

```diff
@@ -0,0 +1,144 @@
+<?php
+namespace StoragelessSession\Http;
+
+use DateTime;
+use Zend\Stratigility\MiddlewareInterface;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Psr\Http\Message\ResponseInterface as Response;
+use Lcobucci\JWT\Token;
+use Lcobucci\JWT\Signer;
+use Lcobucci\JWT\Builder;
+use StoragelessSession\Session\Data;
+use Lcobucci\JWT\Parser;
+use Lcobucci\JWT\ValidationData;
+use Lcobucci\JWT\Signer\Key;
+
+final class SessionMiddleware implements MiddlewareInterface
```

Comment:
You don't need a middleware interface. Stratigility and most other middleware frameworks accepts any callable with `__invoke($req, $res, $next)` signature.